test/CMakeLists.txt

```diff
@@ -18,6 +18,9 @@ if(WITH_SANITIZERS)
 endif()
 
 
+# build test data before running any tests
+add_subdirectory(data)
```

Comment:
The comment is probably misleading, I'm going to change or remove it.
The test data is build during `make` and used during `make test`.
Since the project is mix of compiled and interpreted code, this is probably ok.
Even the `run_tests` binary is compiled during `make`.

Reply:
Well the fact when something is compiled is a different thing from target dependencies. It's too involved for a PR discussion... And I'd need to do some investigation. Might look into it later, no need to block on it now.